### DISCUSSION TITLE
Welcome to radotp Discussions!

### DISCUSSION THREAD

**Discussion by abbas-gheydi** [Abbas-gheydi/radotp#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rusdy21** [comment]
Hi after successfull installed, and can't login using admin/admin

**Reply by abbas-gheydi** [reply]
you must have nginx as a reverse proxy and ssl termination in front of radotp web interface.
cookie setting sets to secure_only that means  it only works on https.
here is example of how to generate SelfSign certs and nginx configs:
``` bash
openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout nginx.key -out nginx.crt
```
nginx config example:
```
server {

listen 443 ssl default_server;
        ssl_certificate      nginx.crt;
        ssl_certificate_key  nginx.key;



location / {

proxy_pass http://127.0.0.1:8080;
}

}
```
open http**s**://SERVER_IP in your browser.

**Reply by rusdy21** [reply]
ok, it works!, anyway we need to configure enable otp for users one by one?

**Comment by abbas-gheydi** [comment]
you can enable Rest API on "radiusd.conf" and connect to it Programmatically.

**Comment by abbas-gheydi** [comment]
The issue has been fixed in version 2. RadOTP now supports HTTPS server natively.

**Comment by Howsyourgulgula** [comment]
Hi, I am receiving an error when trying to pull the image. 
Please advise.

Pulling radotp (ghcr.io/abbas-gheydi/radotp:main) . . .
edaedc954fb5: Pulling fs layer
fcfb138e1122: Pulling fs layer
79183bbdbf3e: Pulling fs layer
8c84ec5d4ada: Waiting
8b6e0f48469d: Waiting
4f4fb700ef54: Waiting
**ERROR: error pulling image configuration: download failed after attempts=1: unknown blob**

**Comment by ehgroehg-git** [comment]
Hi, great product,
 is it possible to use openLdap instead of Windows AD ? Sometimes is needed an embeded solution for authentication without acess to a Windows AD.
I tried but it seems that radotp is not accepting this form of users and openLdap is accepting only this form
-this was with ForceSearchForSamAccountName = true, user is not even queried to openldap
received a radius packet for user: uid=user1, dc=domain
2024/08/20 07:56:38 unexpected input string uid=user1, dc=domain
-with ForceSearchForSamAccountName = false 
received a radius packet for user: user1@domain
password is empty for user:  user1@domain
invalid domain format , user is queried but openldap is not accepting the format user@domain
Thank you

**Reply by badcrc** [reply]
Did you manage to use openldap?

I'm trying the same thing and I cannot validate the users.

**Comment by iamshekoni** [comment]
is it possible to have radotp send the QR Code to the users via email after signing them on the app?

**Comment by iamshekoni** [comment]
@abbas-gheydi thanks for the help
updated the config file and restarted all services but the radotp is stucked on starting. here is the screen shot

![Screenshot 2024-08-21 173509](https://github.com/user-attachments/assets/1fe8c6d1-2db5-46f1-905d-0ba5d5bfb343)
![Screenshot 2024-08-21 173509](https://github.com/user-attachments/assets/4d1a57bf-ff4a-4614-bbb7-b6fdb262599c)

**Reply by abbas-gheydi** [reply]
Hi dear @iamshekoni , I think you should check the RadOTP container log to investigate the source of the problem.

**Reply by iamshekoni** [reply]
thank you for your swift response. i have just noticed i change the 0.0.0.0:1812 in the config file.

**Reply by iamshekoni** [reply]
Thank i have solved the issue

**Comment by iamshekoni** [comment]
Great work i want to make use of Enable_Fortinet_Group_Name please can u assist with the step by step am using it with fortigate firewall
your response will be highly appreciated

i want to grant different permissions to any users assigned to the group vpnuser and vpnadmins

Just advise on what to do in Radotp to make it work and if you can assist with the Fortigate steps too this will be much appreciated.....

I have currently created the groups on AD and Fortigate linked the groups on Fortigate via select. but after testing its only using the vpnadmins only while no user is added on AD. Please assist!!!

**Reply by abbas-gheydi** [reply]
Sure! 
The Enable_Fortinet_Group_Name attribute in FortiGate's RADIUS configuration is used to specify the group name for user authentication. When a user attempts to authenticate, FortiGate checks the group name provided by the RADIUS server against the group names configured on the FortiGate. This helps in applying specific policies and permissions based on the user's group membership.

Here are some useful links for more detailed information and configuration examples:

1. https://docs.fortinet.com/document/fortigate/7.4.0/administration-guide/759080/configuring-a-radius-server
This guide provides a comprehensive overview of setting up a RADIUS server with FortiGate 

2. https://community.fortinet.com/t5/FortiGate/Technical-Tip-How-to-define-group-based-authorization/ta-p/198062
This article explains how to set up group-based authorization using RADIUS on FortiGate 

3. https://docs.fortinet.com/document/fortigate/7.4.2/administration-guide/710485/restricting-radius-user-groups-to-match-selective-users-on-the-radius-server 
This documentation details how to restrict RADIUS user groups to match specific users on the RADIUS server 

In the RADOTP configuration section, make the following changes:

 
1. In the [radius] section, change Enable_Fortinet_Group_Name = false to true.
 
2. In the [ldap] section, add the AD groups to FortiGroups = ["yyy", "xx"] that exactly match the groups you created on FortiGate.(The group names on Active Directory and FortiGate must be exactly the same)Also, uncomment the line by removing the # at the beginning.
then add a user on active to one of those groups.

I hope this helps! If you have any more questions, feel free to ask.

**Reply by badcrc** [reply]
Hi, 

I'm trying to use the group names but I cannot manage to send the group in the RADIUS responde to the Fortinet. Without groups it works perfectly.

<img width="639" alt="Screenshot 2025-02-18 at 11 25 44" src="https://github.com/user-attachments/assets/df68f70f-075f-4d82-9b96-06b3b3b91ca4" />

I'm using [smblds](https://github.com/smblds/smblds-container) for AD/LDAP and also tried with a Windows 2022.

I've the user "koldoa" in the group "group2":

```
# koldoa, Users, domain.local
dn: CN=koldoa,CN=Users,DC=domain,DC=local
objectClass: top
objectClass: person
objectClass: organizationalPerson
objectClass: user
cn: koldoa
givenName: koldoa
instanceType: 4
whenCreated: 20250214125011.0Z
uSNCreated: 4243
name: koldoa
objectGUID:: Vu5wgOG8jEGmQWAj+OwMWg==
badPwdCount: 0
codePage: 0
countryCode: 0
badPasswordTime: 0
lastLogoff: 0
lastLogon: 0
primaryGroupID: 513
objectSid:: AQUAAAAAAAUVAAAAwybGPKkrFy1BoaPLXQQAAA==
accountExpires: 9223372036854775807
logonCount: 0
sAMAccountName: koldoa
sAMAccountType: 805306368
userPrincipalName: koldoa@domain.local
objectCategory: CN=Person,CN=Schema,CN=Configuration,DC=domain,DC=local
uid: koldoa
mail: koldoa@domain.local
pwdLastSet: 133840110110806640
userAccountControl: 512
memberOf: CN=grupo2,DC=domain,DC=local
lastLogonTimestamp: 133840110288539400
whenChanged: 20250214125028.0Z
uSNChanged: 4247
distinguishedName: CN=koldoa,CN=Users,DC=domain,DC=local


# grupo2, domain.local
dn: CN=grupo2,DC=domain,DC=local
objectClass: top
objectClass: group
cn: grupo2
instanceType: 4
whenCreated: 20250214125011.0Z
whenChanged: 20250214125011.0Z
uSNCreated: 4242
name: grupo2
objectGUID:: a8/wEyQD5kOj/szZhhMZtA==
objectSid:: AQUAAAAAAAUVAAAAwybGPKkrFy1BoaPLXAQAAA==
sAMAccountName: $F391A92-BF098F01F296DAE2
sAMAccountType: 268435456
groupType: -2147483646
objectCategory: CN=Group,CN=Schema,CN=Configuration,DC=domain,DC=local
member: CN=koldoa,CN=Users,DC=domain,DC=local
uSNChanged: 4246
distinguishedName: CN=grupo2,DC=domain,DC=local
```


And this is my radotp config:

```
[radius]
	Authentication_Mode = "only_password"
	Enable_Fortinet_Group_Name = true

[ldap]
	FortiGroups = [ "grupo1", "grupo2" ]
	ldapServers = ["smbad"]
	basedn = "DC=domain,DC=local"
	port = 389
	security = 0
	ForceSearchForSamAccountName = true
```

And this is the log of radotp:

```
2025/02/18 10:31:52 received a radius packet for user: koldoa
2025/02/18 10:31:52 Access-Accept to 192.168.229.1:20772 for koldoa stage active directory
```

**Reply by badcrc** [reply]
@abbas-gheydi I got it working but only works in `two_fa_optional_otp` and `two_fa` modes, doesn't work in `only_password`, could it be a bug?

**Reply by abbas-gheydi** [reply]
Hi dear @badcrc 
Okay, I'll check it and let you know.
Thank you a lot!

**Comment by iamshekoni** [comment]
Dear Abbas-Gheydi/Radotp

Thank you for your response. We appreciate your response. It was very
helpful and an eye opener to us.

**Comment by iamshekoni** [comment]
Dear Abbas-Gheydi/Radotp

Thank you for your usual support, please can you consider the following in your next release

1. Enrollment Barcode to be send to the users email to increase confidentiality (Field to input the users email at the point of enrolling user for OTP)
2. Creating other Users and adding them to user or administrator roles
3. Prevent barcode reuse
4. Company logo upload
5. Send OTP Code via email too
6. How can we use it to authenticate for windows and other applications?

This will be a great feature to have. God bless you for the good work.
i won't mind to have a personal connection with you also if you don't mind.
Cheers

**Reply by abbas-gheydi** [reply]
Dear @iamshekoni ,

I sincerely appreciate your thoughtful suggestions for improving radotp. Your insights are invaluable.

Additionally, you can find my contact information on my GitHub profile. I look forward to expanding the connections.

Thank you once again.

**Comment by iamshekoni** [comment]
No I could not move forward since then.

**Reply by badcrc** [reply]
Finally I'm using https://github.com/smblds/smblds-container and seems to work :)
